app/controllers/comments_controller.rb

```diff
@@ -87,18 +87,26 @@ def check_if_restricted
     redirect_to new_user_session_path(restricted_commenting: true)
   end
 
-  # Check to see if the ultimate_parent is a Work, and if so, if it allows
+  # Check to see if the ultimate_parent is a Work or AdminPost, and if so, if it allows
   # comments for the current user.
   def check_parent_comment_permissions
     parent = find_parent
-    return unless parent.is_a?(Work)
+    if parent.is_a?(Work)
+      translation_key = "work"
+      parent_path = work_path(parent)
+    elsif parent.is_a?(AdminPost)
+      translation_key = "admin_post"
+      parent_path = admin_post_path(parent)
+    else
+      return
+    end
 
     if parent.disable_all_comments?
-      flash[:error] = ts("Sorry, this work doesn't allow comments.")
-      redirect_to work_path(parent)
+      flash[:error] = t("comments.commentable.permissions.#{translation_key}.disable_all")
+      redirect_to parent_path
```

Comment:
I think `redirect_to parent` would return absolute URLs unless we pass `only_path: true` as options. I like `polymorphic_path`.

Reply:
For my own edification, what's the reason for trying to avoid absolute URLs?

From rails [docs](https://api.rubyonrails.org/classes/ActionDispatch/Routing/PolymorphicRoutes.html) seems `redirect_to`uses `polymorphic_url` under the hood anyway.

Reply:
I was thinking of [AO3-4630](https://otwarchive.atlassian.net/browse/AO3-4630) (absolute URLs are less friendly to full-page nginx caching). However. on the same issue:
> Note that `redirect_to` also uses absolute URLs by default, but we don't need to hunt for their uses because we rewrite redirects with nginx (to fix [AO3-5945](https://otwarchive.atlassian.net/browse/AO3-5945)).

So `redirect_to parent` should be good.

Reply:
Absolute URLs can also cause issues in JS-based tests, because the host used to generate all URLs is `example.com`, so poltergeist/phantomjs ends up looking up actual pages from `example.com`. But I don't think it makes a difference outside of the test environment.